src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingRepairService.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.courtsentencing
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data.MigrationContext
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomisprisoner.model.PrisonerId
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationType
+import java.time.LocalDateTime
+import kotlin.String
+
+@Service
+class CourtSentencingRepairService(
+  val courtSentencingMigrationService: CourtSentencingMigrationService,
+  val telemetryClient: TelemetryClient,
+) {
+  suspend fun resynchronisePrisonerCourtCases(offenderNo: String) {
+    courtSentencingMigrationService.synchronisePrisonerCases(
+      offenderNo = offenderNo,
+      deleteExisting = true,
+      context = MigrationContext(
+        type = MigrationType.COURT_SENTENCING,
+        migrationId = LocalDateTime.now().toString(),
```

Comment:
a pragmatic compromise here - pretend there is a migration ID so we don't have to make big changes around the retry mapping functionality that assumes we have a migrationId just so it can log it